src/lib/party.ts

```diff
@@ -657,8 +657,14 @@ class Party {
   }
 
   hash() {
-    let fields = Types.Party.toFields(toPartyUnsafe(this));
-    return Ledger.hashPartyFromFields(fields);
+    // these two ways of hashing are (and have to be) consistent / produce the same hash
```

Comment:
this is what fixes the new example. the previous version was always returning variables, even outside the circuit